src/bindgen/ir/cfg.rs

```diff
@@ -166,139 +198,166 @@ impl Cfg {
-
-                    match key {
-                        DefineKey::Boolean(..) => {}
-                        DefineKey::Named(key_name, key_value) => {
-                            if cfg_name == key_name && cfg_value == key_value {
-                                return true;
-                            }
-                        }
-                    }
+            Cfg::Named(cfg_name, cfg_value) => {
+                let define = config.defines.iter().find(|(key, ..)| {
+                    DefineKey::Named(&cfg_name, &cfg_value) == DefineKey::load(key)
+                });
+                if let Some((_, define)) = define {
+                    Some(Condition::Named(define.to_owned()))
+                } else {
+                    warn!(
+                        "Missing `[defines]` entry for `{}` in cbindgen config.",
+                        Cfg::Named(cfg_name, cfg_value)
+                    );
+                    None
+                }
+            }
+            Cfg::Any(children) => {
+                let conditions: Vec<_> = children
+                    .into_iter()
+                    .filter_map(|x| x.to_condition(config))
+                    .collect();
+                match conditions.len() {
+                    0 => None,
+                    1 => conditions.into_iter().next(),
+                    _ => Some(Condition::Any(conditions)),
                 }
-
-                false
             }
-            &Cfg::Any(ref cfgs) => cfgs.iter().all(|x| x.has_defines(config)),
-            &Cfg::All(ref cfgs) => cfgs.iter().all(|x| x.has_defines(config)),
-            &Cfg::Not(ref cfg) => cfg.has_defines(config),
+            Cfg::All(children) => {
+                let cfgs: Vec<_> = children
+                    .into_iter()
+                    .filter_map(|x| x.to_condition(config))
+                    .collect();
+                match cfgs.len() {
+                    0 => None,
+                    1 => cfgs.into_iter().next(),
+                    _ => Some(Condition::All(cfgs)),
+                }
+            }
+            Cfg::Not(child) => child
+                .to_condition(config)
+                .map(|cfg| Condition::Not(Box::new(cfg))),
         }
     }
+}
 
-    fn write_condition<F: Write>(&self, config: &Config, out: &mut SourceWriter<F>) {
-        match self {
-            &Cfg::Boolean(ref cfg_name) => {
-                let mut define: &str = cfg_name;
-
-                for (key, define_value) in &config.defines {
-                    let key = DefineKey::load(key);
-
-                    match key {
-                        DefineKey::Boolean(key_name) => if cfg_name == key_name {
-                            define = define_value;
-                        },
-                        DefineKey::Named(..) => {}
-                    }
-                }
+#[derive(Debug, Clone)]
+pub enum Condition {
+    Boolean(String),
+    Named(String),
```

Comment:
Is there any difference between `Boolean` and `Named`?

Reply:
No. I merged them into `Condition::Define`. :)